### PR TITLE
fixing degenerated onNext method in Requester automatons

### PR DESCRIPTION
**Pull request by lehecka**
the onNext method name was very confusing, colliding with ReactiveStreams onNext.
It was also historically accidently implemented as async method (scheduling a callback on the executor).
This PR fixes that problem and removed the confusion in the situation when we are processing the initial payload.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -203,7 +203,8 @@ add_executable(
   test/ReactiveSocketResumabilityTest.cpp
   test/SmartPointersTest.cpp
   test/AllowanceSemaphoreTest.cpp
-  test/ResumeIdentificationTokenTest.cpp)
+  test/ResumeIdentificationTokenTest.cpp
+  test/PayloadTest.cpp.cpp)
```
Comment (yschimke) [inline CMakeLists.txt]: too many .cpp

File: src/automata/RequestResponseRequester.cpp [hunk 2]
```diff
@@ -110,31 +104,35 @@ void RequestResponseRequester::onNextFrame(Frame_ERROR&& frame) {
...
-      //       after receiving it. Add unit tests for this.
-      if (frame.header_.flags_ & FrameFlags_COMPLETE) {
-        state_ = State::CLOSED;
-        end = true;
-      }
+      state_ = State::CLOSED;
       break;
     case State::CLOSED:
+      // should not be receiving frames when closed
+      // if we ended up here, we broke some internal invariant fo the class
```
Comment (yschimke) [inline src/automata/RequestResponseRequester.cpp]: typo: fo

File: src/automata/RequestResponseRequester.cpp [hunk 3]
```diff
@@ -110,31 +104,35 @@ void RequestResponseRequester::onNextFrame(Frame_ERROR&& frame) {
 }
 
 void RequestResponseRequester::onNextFrame(Frame_RESPONSE&& frame) {
-  bool end = false;
   switch (state_) {
     case State::NEW:
       // Cannot receive a frame before sending the initial request.
       CHECK(false);
```
Comment (yschimke) [inline src/automata/RequestResponseRequester.cpp]: Are these absolutely only internal invariants? e.g. data corruption on the wire could end up here?

Comment (lehecka) [inline src/automata/RequestResponseRequester.cpp]: yes, if end up failing in the CHECK, we broke the invariant and didn't handle some cases. The frames sent over the wire should never cause the check fail.

**Pull request merged**